catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java

```diff
@@ -89,12 +92,7 @@ public SaxEventHandlerDelegate(List<SaxEventHandler> eventHandlers) {
      * populated with all the {@link Attribute}s parsed by the {@link SaxEventHandlerDelegate#eventHandlers}
      * @throws CatalogTransformerException
      */
-    public Metacard read(InputStream inputStream) throws CatalogTransformerException {
-
-        /*
-         * Create a new MetacardImpl with the proper MetacardType
-         */
-        Metacard metacard = new MetacardImpl(metacardType);
+    public void read(InputStream inputStream) throws CatalogTransformerException {
```

Comment:
if you change this to return a `SaxEventHandlerDelegate`, you can chain the calls. I don't know if it buys you anything, but smooths the code out a bit.